jwst/residual_fringe/residual_fringe.py

```diff
@@ -134,7 +132,7 @@ def do_correction(self):
 
         self.background_fit = np.zeros(self.input_model.data.shape)
         self.knot_locations = np.empty(self.input_model.data.shape)
```

Comment:
```suggestion
        self.knot_locations = np.full_like(self.input_model.data, np.NaN)
```
This is one way to create an array of that shape filled with `np.NaN` values.